feat: use docker commands for stack handling (DEV-1530) (#261)
dasch-swiss/dsp-tools: docs/dsp-tools-usage.md

## Start a DSP stack on your local machine 

DSP-API is the heart of the DaSCH software - it is a server application for storing data from the Humanities.
DSP-APP is our generic user interface, i.e. the viewer to look at the data. It's a server application, too.

For testing purposes, it is sometimes necessary to run DSP-API and DSP-APP on a local machine. But the startup 
and shutdown of API and APP can be complicated: Both repos need to be cloned locally, a `git pull` has to be executed 
from time to time to stay up to date, and then there are several commands for each repository to remember. And your 
local clone of the API might get cluttered with old files over time.

> Another challenge is the software that DSP depends upon, and that should be kept up to date: JDK, node, npm, Angular, 

Reviewer: ```suggestion
Another challenge is the software that DSP depends on, and that should be kept up to date: JDK, node, npm, Angular, 
```

etc. And it might happen that a dependency is replaced, e.g. JDK 11 Zulu by JDK 17 Temurin. A non-developer can quickly 

> get lost in this jungle. 

Reviewer 2: I recommend to remove this whole section from the documentation. It is not the right place to inform the user about decisions that were made regarding the setup. I assume the user just wants to know what to do - as briefly and simple as possible. 
We also have the problem in DSP-API that different "users" are targeted by the same documentation (end-users, internal/external software developers). This resulted in a messy documentation over time. We can brainstorm about this topic, if you like. It's not easy to set this up alone.

Author: I principally agree with you, but there is one reason why I am reluctant: It is now a years-old practice to install all this software on the machines of new employees, and to execute commands in the repositories. For me it was an "aha moment" to learn the distinction between building from the repositories (needs dependencies) versus running Docker containers (no dependencies needed). That's the reason why I feel the need to clearly document this change. If I just communicate it internally in a chat, it will be forgotten after some days/weeks. 

But I'm open for brainstorming about better options.

That's why dsp-tools offers a command to facilitate the handling of API and APP:
```

> dsp-tools start-stack

Reviewer: maybe I'm professionally deformed by our make targets... but to me `dsp-tools stack-up` (or even shorter `dsp-tools stack`) and `stack-down` would sound more familiar/intuitive.

Author: I wanted to find a terminology that prevents confusion with the make-targets, otherwise I confuse them. In the initial phase, when the two were similar, I accidentally typed things like `dsp-tools stack-down-delete-volumes`.

```

> This calls Docker commands to start up the latest deployed versions of API and the APP, i.e. the versions that are 

Reviewer 2: ```suggestion
This calls Docker commands to start up the latest versions of DSP-API and DSP-APP, i.e. the versions that are 
```

> running on [https://admin.dasch.swiss](https://admin.dasch.swiss). It's no longer necessary to have two local clones, 

Reviewer 2: ```suggestion
running on [https://admin.dasch.swiss](https://admin.dasch.swiss).
```

nor to execute `make` commands inside them. The only requirement for this command is that Docker must be running. It 

> can even be executed on a Windows machine without any of the software dependencies.

Reviewer: maybe I just overlooked it... but you'd still have docker installed&running as a requirement - so that should definitely be explicitly mentioned.

Reviewer: also, speaking of requirements: do we write anywhere that python (and which python version) must be installed? just came to my mind...

Author: dsp-tools is a pip package, so it's difficult to have it without Python. setup.py says `python_requires=">=3.9.0"`, so we already have it pinned there. But actually, I don't think that dsp-tools really requires a certain Python version, does it?

Reviewer: yes... don't ask me why... but I remember Eric at some point having an issue which turned out to be him having 3.8 installed. So probably it would be good to specify that somewhere as I don't think pip will stop you from installing it on a lower python version (not sure though). But that's by no means urgent.

> When your work is done, shut down API and APP with

Reviewer 2: ```suggestion
Shut down DSP-API and DSP-APP:
```

```
dsp-tools stop-stack
```

> This deletes all Docker volumes, and removes all data that was in the database.

Reviewer 2: ```suggestion
This command deletes all Docker volumes, and removes all data that was in the database.
```

Some notes:

 - As long as you want to keep the data in the database, don't execute `dsp-tools stop-stack`. It is possible to 

>    leave the API up for a long time. If you want to save power, you can pause Docker. When you resume it, the API 

Reviewer 2: ```suggestion
   leave the API up for a long time. If you want to save resources, you can pause Docker. When you resume it, the API 
```

   will still be running, in the state how you left it.
 - You can also send your computer to sleep while the DSP stack is running. For this, you don't even need to pause 
   Docker.

>  - This command provides you with the latest released & deployed version of API and APP. If you want to run a 

Reviewer 2: ```suggestion
 - This command provides you with the latest versions of DSP-API and DSP-APP. If you want to run a 
```
